src/Http/Controllers/Support/ResolveController.php

```diff
@@ -49,103 +67,203 @@ class ResolveController extends Controller
+        $names = CorporationInfo::whereIn('corporation_id', $ids->flatten()->toArray())
+            ->get()
+            ->map(function ($corporation) {
+                return collect([
+                    'id' => $corporation->corporation_id,
+                    'name' => $corporation->name,
+                ]);
             });
 
-            $chunk = $chunk->filter(function ($id) use ($names) {
-
-                return ! $names->contains('factionID', $id);
-            });
+        return $this->cacheIDsAndReturnUnresolvedIDs($names, $ids);
+    }
 
-            // quick break if no more IDs must be resolve by ESI
-            if ($chunk->count() == 0)
-                return;
+    private function resolveIDsfromESI($ids, $eseye)
+    {
 
-            // Finally, grab outstanding ids and resolve their names
-            // using Esi.
+        // Finally, grab outstanding ids and resolve their names
+        // using Esi.
 
+        try {
             $eseye->setVersion('v2');
-            $eseye->setBody($chunk->flatten()->toArray());
+            $eseye->setBody($ids->flatten()->toArray());
             $names = $eseye->invoke('post', '/universe/names/');
 
-            collect($names)->each(function ($name) use (&$response) {
+            collect($names)->each(function ($name) {
 
                 // Cache the name resolution for this id for a long time.
                 cache([$this->prefix . $name->id => $name->name], carbon()->addCentury());
+                $this->response[$name->id] = $name->name;
 
-                $response[$name->id] = $name->name;
             });
+
+        } catch (\Exception $e) {
+            // If this fails split the ids in half and try to self referential resolve the half_chunks
+            // until all possible resolvable ids has processed.
+            if ($ids->count() === 1) {
+                // return a singleton unresolvable id as 'unknown'
+                $this->response[$ids->first()] = trans('web::seat.unknown');
+            } else {
+                //split the chunk in two
+                $half = ceil($ids->count() / 2);
+                //keep on processing the halfs independently,
+                //ideally one of the halfs will process just perfect
+                $ids->chunk($half)->each(function ($half_chunk) use ($eseye) {
+
+                    //this is a selfrefrencial call.
+                    $this->resolveIDsfromESI($half_chunk, $eseye);
+                });
+            }
+
+        }
+
+    }
+
+    /**
+     * @param \Illuminate\Support\Collection $names
+     * @param \Illuminate\Support\Collection $ids
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    private function cacheIDsAndReturnUnresolvedIDs(Collection $names, Collection $ids) : Collection
+    {
+        $names->each(function ($name) {
+
+            cache([$this->prefix . $name['id'] => $name['name']], carbon()->addCentury());
```

Comment:
Why this long and not just forever?

Reply:
i simply reused the caching period that was used before: https://github.com/eveseat/web/blob/master/src/Http/Controllers/Support/ResolveController.php#L114 

i thought you got your reasons not to cache it for forever